### ISSUE TITLE
"Too bad. Something's wrong ;-(."

### ISSUE THREAD

**Issue by Lesik** [tranxuanthang/villain3ds#9]
Hey,

I keep getting this error message after the download:
```
Too bad. Something's wrong ;-(.
"/home/me/Villain3DS/make_cdn_cia"
"/home/me/Villain3DS/raw/0004000000030700"
"/home/me/Villain3DS/cias/0004000000030700.cia"
```

After the download, both the raw and cias folders are emptied. The `make_cdn_cia` is 0 bytes.

_Edit_: I am running Linux. I have set up a Windows VM and can confirm that it works on Windows.

**Comment by tranxuanthang** [comment]
Maybe it failed to download make_cdn_cia file the first time app ran, but still left a empty 0kB file at its location. So next times, the app sees that the make_cdn_cia is still there and considers it already downloaded from before.
So, first, let's try to delete the make_cdn_cia file, then try to download some titles with the app again, to make it redownload the make_cdn_cia. If it doesn't work, download make_cdn_cia [here](https://github.com/tranxuanthang/villain3ds/raw/master/make_cdn_cia) and put it to your Villain3DS directory. Remember to chmod +x to make the file executable too. I'll look more into this issue later.

**Comment by Lesik** [comment]
I let it re-download `make_cdn_cia`, but the execution bit was not set automatically. After setting it, it works now.

Some considerations:
- setting execution bit doesn't seem to work on Linux:
https://github.com/tranxuanthang/villain3ds/blob/a29fedccfa4a90cfb424696a3bd7a46e226613c7/download2.js#L169
- check if `make_cdn_cia` is 0 bytes, or check checksum
- don't delete raw files until after `make_cdn_cia` completed _successfully_

**Comment by tranxuanthang** [comment]
I don't get the execution bit part. Could you explain more about it?
And I'll try to do the 2nd and 3rd bullets tonight.

**Comment by Lesik** [comment]
I meant that the linked LOC is, I'm not a JS dev but I think so, supposed to set the execution bit (basically "`chmod +x`") on the `make_cdn_cia` executable. However, this doesn't seem to work on my machine — the execution bit is not set after removing the `~/Villain3DS` folder and letting the software redownload the executable.

On a different note, I think I forgot to say thanks for the work you're putting in this project.

**Comment by Lesik** [comment]
Mhh, scratch that. For some reason I read `{'flags': 'w'}` as `{'flags': 'x'}` and interpreted that as setting the execution bit. Probably should put my hands away from JS. :)

**Comment by tranxuanthang** [comment]
I see, the flag is for the mode using to open the file. 'x' means open file for writing only, pointer at beginning of the file. Read more here http://php.net/manual/en/function.fopen.php
I think I actually forgot to chmod +x for the make_cdn_cia file. The old download script has it.
https://github.com/tranxuanthang/villain3ds/blob/a29fedccfa4a90cfb424696a3bd7a46e226613c7/renderer-download.js#L477

**Comment by tranxuanthang** [comment]
Fixed in 0.1.2
